contribs/gmf/src/directives/search.js

```diff
@@ -0,0 +1,281 @@
+
+
+
+/**
+ * @constructor
+ * @param {angular.Scope} $rootScope Angular root scope.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {ngeo.CreateGeoJSONBloodhound} ngeoCreateGeoJSONBloodhound The ngeo
+ * create GeoJSON Bloodhound service.
+ * @ngInject
+ */
+gmf.SearchController = function($rootScope, $compile,
+    ngeoCreateGeoJSONBloodhound) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.rootScope_ = $rootScope;
+
+  /**
+   * @type {angular.$compile}
+   * @private
+   */
+  this.compile_ = $compile;
+
+  /**
+   * @type {ngeo.CreateGeoJSONBloodhound}
+   * @private
+   */
+  this.ngeoCreateGeoJSONBloodhound_ = ngeoCreateGeoJSONBloodhound;
+
+  var map = this['getMapFn']();
+  goog.asserts.assertInstanceof(map, ol.Map);
+
+  /**
+   * @type {!ol.Map}
+   * @export
+   */
+  this.map = map;
+
+  /**
+   * @type {ol.layer.Vector}
+   * @private
+   */
+  this.vectorLayer_ = this.createVectorLayer_();
+
+  /**
+   * TODO must be definable by an integrator
+   * @type {Array<string>}
+   * @private
+   */
+  this.datasources_ = [{
+    dataTitle: 'From demo 1.6',
+    url: 'http://geomapfish-demo.camptocamp.net/1.6/wsgi/fulltextsearch'
+  }, {
+    dataTitle: 'From Cartoriviera - ',
+    url: 'http://map.cartoriviera.ch/main/wsgi/fulltextsearch',
+    groupByProperties: ['04rue', '03adresse']
+  }];
+
+  /**
+   * @type {TypeaheadOptions}
+   * @export
+   */
+  this.options = /** @type {TypeaheadOptions} */ ({
+    highlight: true,
+    hint: undefined,
+    minLength: undefined
+  });
+
+  /**
+   * @type {Array.<TypeaheadDataset>}
+   * @export
+   */
+  this.datasets = [];
+  for (var i = 0; i < this.datasources_.length; i++) {
+    var datasource = this.datasources_[i];
+    var groupByProperties = datasource['groupByProperties'] || [];
+    var filter;
+    do {
+      //Add an optional filter function to keep objects only from one
+      //"layername" from a GMF's fulltextsearch service.
+      var dataTitle = datasource['dataTitle'];
+      if (groupByProperties[0]) {
+        filter = this.filterLayername_(groupByProperties[0]);
+        dataTitle = dataTitle + groupByProperties[0];
+        groupByProperties.shift();
+      } else {
+        filter = null;
+      }
+
+      this.datasets.push(this.createDataset_(datasource['url'],
+                                             filter,
+                                             dataTitle));
+    }
+    while (groupByProperties.length > 0);
+  }
+
+
+
+  /**
+   * @type {ngeox.SearchDirectiveListeners}
+   * @export
+   */
+  this.listeners = /** @type {ngeox.SearchDirectiveListeners} */ ({
+    selected: angular.bind(this, gmf.SearchController.selected_)
+  });
+
+};
+
+
+/**
+ * @return {ol.layer.Vector} The vector layer.
+ * @private
+ */
+gmf.SearchController.prototype.createVectorLayer_ = function() {
+  var vectorLayer = new ol.layer.Vector({
+    source: new ol.source.Vector()
```

Comment:
See https://github.com/openlayers/ol3/blob/master/src/ol/interaction/drawinteraction.js#L276-L279 for options used for unmanaged layers.